util/failover.go

```diff
@@ -190,12 +189,17 @@ func ReplicationStatus(request ReplicationStatusRequest) (ReplicationStatusRespo
 		}
 
 		// get the instance name that is recognized by the Operator, which is the
-		// first part of the name
-		instance.Name = instanceNamePattern.FindStringSubmatch(rawInstance.PodName)[1]
-
-		// get the node that the replica is on based on the "replica name" for this
-		// instance
-		instance.Node = instanceNodeMap[instance.Name]
+		// first part of the name and is kept on a deployment label. We have these
+		// available in our instanceNodeMap, and because we skip over the primary,
+		// this will not lead to false positive
+		//
+		// This is not the cleanest way of doing it, but it works
+		for name, node := range instanceNodeMap {
+			if strings.HasPrefix(rawInstance.PodName, name) {
+				instance.Name = name
+				instance.Node = node
```

Comment:
Do you need to `break` out of the loop here given that you've found the instance?

Reply:
Yes.